src/qibolab/instruments/simulator/emulator_test/ibmfakebelem_q01/parameters.json

```diff
@@ -0,0 +1,226 @@
+{
```

Comment:
Are these paramerers used for creating any actual `Platform` for emulation?
I would suggest converting this to an actual `Platform` following the directory structure of https://github.com/qiboteam/qibolab_platforms_qrc:
```
ibmfakebelem_q01
    platform.py
    parameters.json
    model_params.yml (potentially renamed to model.json)
```
where `platform.py` contains a `create()` method that returns the `Platform` object with the simulator instrument.

This should also allow you to test with qibocal.

Reply:
Sounds good. Where should I store each of these platform directories? Should new emulator platforms be added to [qibolab_platforms_qrc](https://github.com/qiboteam/qibolab_platforms_qrc): as well? I will probably keep the minimum here for pytests, perhaps in a new directory src/qibolab/emulator similar to src/qibolab/dummy?

Reply:
I missed that there was already a conversation, so let me repeat here as well `emulators/` in Qibolab repo should be fine.

These are not strictly related to the QRC, so I would not put them there.
Rather, I'd upgrade the meaning of the `QIBOLAB_PLATFORMS` variable to be `PATH`-like (i.e. a `:` separated list, instead of a single path).

Reply:
For the time being, it is assumed that that variable is a path to a single directory:
https://github.com/qiboteam/qibolab/blob/5811117b0f0d913957539d854aefeee5499c2c87/src/qibolab/__init__.py#L22-L30

For testing, you could just add the one you care about (i.e. just the emulators). But to make convenient using both together, and switching from one to another, a minor patch is needed (to upgrade it to a list).

Reply:
I am not sure if we need an `emulators/` directory to be shipped with the qibolab package. If we are only using these for testing, I think it makes more sense to put them in `tests/` like we do with other drivers (see `tests/dummy_qrc`) and only keep the emulator primitives inside the package, so that users can define their own emulators. But I am fine either way.

The patch for converting `QIBOLAB_PLATFORMS` is also fine, but can (and probably should) be implemented in another small PR as it is not related to emulators (I may also want to have two different paths where I define lab platforms).

Reply:
For the test related emulator platforms, I am happy to put them under `tests/emulator_qrc`, similar to `tests/dummy_qrc`.

For the actual emulator platforms, if I understood both of you correctly, @alecandido is suggesting to put them under `emulators/` directory while @stavros11 prefers not to have them as part of the qibolab package. At the same time @alecandido does not think they belong to qibolab_platforms_qrc. So where should they be stored in? Perhaps a separate qibolab_emulators repo?

Based on the discussion so far, when an emulator for a specific device (e.g. QPU1) is properly set up, we would have two platform folders: QPU1 and QPU1_emulator. The runcard file `parameters.json` present in both of them should in principle be identical, but the QPU1 and QPU1_emulator folders would be stored in different directories. Is that the arrangement?

Reply:
> Based on the discussion so far, when an emulator for a specific device (e.g. QPU1) is properly set up, we would have two platform folders: QPU1 and QPU1_emulator. The runcard file `parameters.json` present in both of them should in principle be identical, but the QPU1 and QPU1_emulator folders would be stored in different directories. Is that the arrangement?

In that case, I think it is fine, and maybe even more convenient, to have them in the same place. If the QPU is hosted in TII QRC labs, that would be the qibolab_platforms_qrc repository. If it is hosted in a different lab some other place (depending on the lab). If parameters.json is shared, we could even think of a better way to load it, like having both the emulator and the hardware platform in the same directory but with different `create` methods. However, I would leave that simplification for later and go with what you are proposing: qpu1, qpu1_emulator in the same parent directory (and parameters.json duplicated).

I think the main issue with the platforms that are commited in PR is that they are not related to QRC QPUs. This may be why @alecandido was suggesting to not put them in qibolab_platforms_qrc (but he can correct me if I am wrong).

Personally, I am not concerned about where we store the platforms, but I would like to keep the qibolab package (the one that goes to pypi) clean of platforms and code that is specific to a particular lab, since our goal is to keep it platform and lab agnostic. We should provide examples on how to write emulator platforms in the docs and also have some examples that are used for testing (tests/dummy_qrc) but after that I would let users define their own platforms and store them wherever they want, even if that means copying a cell defining a platform from our docs. Of course this is my opinion and I am open to better suggestions.

Reply:
> I think the main issue with the platforms that are commited in PR is that they are not related to QRC QPUs. This may be why @alecandido was suggesting to not put them in qibolab_platforms_qrc, to which I also agree (but he can correct me if I am wrong).

I confirm :)

> Personally, I am not concerned about where we store the platforms, but I would like to keep the qibolab package (the one that goes to pypi) clean of platforms and code that is specific to a particular lab, since our goal is to keep it platform and lab agnostic.

I fully agree about this.

> . We should provide examples on how to write emulator platforms in the docs and also have some examples that are used for testing (tests/dummy_qrc) but after that I would let users define their own platforms and store them wherever they want, even if that means copying a cell defining a platform from our docs.


Personally, I consider copying from examples perfectly fine, and a good learning practice. You copy them, and start customizing for your needs.
The moment you're familiar with the toolbox, then you will be encouraged to write it from scratch (while if there are blocks that have to be repeated in any case, those should be part of the toolbox).

Reply:
Added only test emulator platforms in e035cb3